### PR TITLE
feat(php-nextgen-client): add support for streaming

### PR DESCRIPTION
**Pull request by neilime**
**php-nextgen** client:

Add support for streaming file (https://spec.openapis.org/oas/latest.html#considerations-for-file-uploads)
Basically returning " Psr\Http\Message\StreamInterface" instead of "SplFileObject"

- Adds a new option "supportStreaming" (false by default to do not add breaking change)

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

- @jebentier
- @dkarlovi
- @mandrean
- @jfastnacht
- @ybelenko
- @renepardon

### CODE REVIEW

**Comment by neilime** [comment]
Hi,

@jebentier, @dkarlovi, @mandrean, @jfastnacht, @ybelenko, @renepardon

Are any of you available for a review?

Thanks

File: modules/openapi-generator/src/test/java/org/openapitools/codegen/php/PhpNextgenClientCodegenTest.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright 2018 OpenAPI-Generator Contributors (https://openapi-generator.tech)
+ * Copyright 2018 SmartBear Software
```
Comment (wing328) [inline modules/openapi-generator/src/test/java/org/openapitools/codegen/php/PhpNextgenClientCodegenTest.java:3]: for new files, I think this line can be removed.

**Comment by wing328** [comment]
can you please review the test failures?

i could repeat the issue locally when testing the output
```
[INFO]
[INFO] Results:
[INFO]
[ERROR] Failures:
[ERROR] org.openapitools.codegen.php.PhpNextgenClientCodegenTest.testInitialConfigValues
[ERROR]   Run 1: PhpNextgenClientCodegenTest.testInitialConfigValues:31 expected [false] but found [null]
[ERROR]   Run 2: PhpNextgenClientCodegenTest.testInitialConfigValues:31 expected [false] but found [null]
[ERROR]   Run 3: PhpNextgenClientCodegenTest.testInitialConfigValues:31 expected [false] but found [null]
[INFO]
[ERROR] org.openapitools.codegen.php.PhpNextgenClientCodegenTest.testSettersForConfigValues
[ERROR]   Run 1: PhpNextgenClientCodegenTest.testSettersForConfigValues:41 expected [true] but found [null]
[ERROR]   Run 2: PhpNextgenClientCodegenTest.testSettersForConfigValues:41 expected [true] but found [null]
[ERROR]   Run 3: PhpNextgenClientCodegenTest.testSettersForConfigValues:41 expected [true] but found [null]
[INFO]
[INFO]
[ERROR] Tests run: 3096, Failures: 2, Errors: 0, Skipped: 0

```

**Comment by neilime** [comment]
@wing328 I've fixed you feedbacks, can you review it again please.

Thanks

**Comment by wing328** [comment]
https://github.com/OpenAPITools/openapi-generator/actions/runs/10062528767/job/27842785278?pr=19192

please update the samples when you've time

**Comment by neilime** [comment]
@wing328 it's done (I've updated the branch with master in the same time)

**Comment by wing328** [comment]
looks good. let's give it a try.

thanks for the PR

**Pull request merged**